src/foundry/foundry.js/collections/documentCollection.d.ts

```diff
@@ -6,7 +6,7 @@ declare global {
    * An abstract subclass of the Collection container which defines a collection of Document instances.
    */
   abstract class DocumentCollection<T extends DocumentConstructor, Name extends string> extends foundry.utils
```

Comment:
Did you intentionally not do it for `set`?

Also I'm not sure I agree with doing it for this class (and the subclasses). What's preventing somebody from creating their own `DocumentCollection` classes that deal with temporary documents? sure, it's not the most usual use case but still... Similar for `CompendiumCollection` and `WorldCollection` (while I agree that it makes more sense for those...)

Reply:
No, `set` just slip through :-|

I also thought about it. My initial thought were, to just apply the `StoredDocument` to `CompendiumCollection` and `WorldCollection` (which would require `TemporaryDocument` in some places, but that's fine).
But in the end there were two methods which tipped the balance. The `set` method, which uses the document id and the `updateAll` method, which throws an error if you have any element in the collection which has no `_id` set.
This brings me to the conclusion that there is some kind of contract that every document in a document collection has an id.

Of course, someone might break this contract. It's JavaScript, you can break nearly everything. But I do not think that it is the purpose of the types to allow such things.